app/src/main/java/protect/budgetwatch/MainActivity.java

```diff
@@ -221,69 +206,72 @@ private void displayAboutDialog()
-            String.format(getString(R.string.app_libraries), appName, libs.toString()) +
-            "</p><hr/><p>" +
-            String.format(getString(R.string.app_resources), appName, resources.toString());
+                "<meta http-equiv=\"content-type\" content=\"text/html; charset=utf-8\" />" +
+                        "<img src=\"file:///android_res/mipmap/ic_launcher.png\" alt=\"" + appName + "\"/>" +
+                        "<h1>" +
+                        String.format(getString(R.string.about_title_fmt),
+                                "<a href=\"" + getString(R.string.app_webpage_url)) + "\">" +
+                        appName +
+                        "</a>" +
+                        "</h1><p>" +
+                        appName +
+                        " " +
+                        String.format(getString(R.string.debug_version_fmt), version) +
+                        "</p><p>" +
+                        String.format(getString(R.string.app_revision_fmt),
+                                "<a href=\"" + getString(R.string.app_revision_url) + "\">" +
+                                        getString(R.string.app_revision_url) +
+                                        "</a>") +
+                        "</p><hr/><p>" +
+                        String.format(getString(R.string.app_copyright_fmt), year) +
+                        "</p><hr/><p>" +
+                        getString(R.string.app_license) +
+                        "</p><hr/><p>" +
+                        String.format(getString(R.string.app_libraries), appName, libs.toString()) +
+                        "</p><hr/><p>" +
+                        String.format(getString(R.string.app_resources), appName, resources.toString());
 
 
         wv.loadDataWithBaseURL("file:///android_res/drawable/", html, "text/html", "utf-8", null);
         new AlertDialog.Builder(this)
-            .setView(wv)
-            .setCancelable(true)
-            .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener()
-            {
-                public void onClick(DialogInterface dialog, int which)
-                {
-                    dialog.dismiss();
-                }
-            })
-            .show();
+                .setView(wv)
+                .setCancelable(true)
+                .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialog.dismiss();
+                    }
+                })
+                .show();
     }
 
-    private void startIntro()
-    {
+    private void startIntro() {
         Intent intent = new Intent(this, IntroActivity.class);
         startActivity(intent);
     }
+
+    private void startCreatePassword() {
+        Intent intent = new Intent(this, PasswordCreationActivity.class);
+        startActivity(intent);
+    }
+
+    private void startAuthorization() {
+        authorized = true;
```

Comment:
How is the main activity receiving a response from the authentication activity that the authentication was successful? The expectation here is that the authentication activity will not "return" until a password is entered, however this is not the case. A better pattern may be to startActivityForResult(). See documentation here:

https://developer.android.com/training/basics/intents/result

That will allow the return result from the authentication activity to be captured, and a more informed decision made about authentication success or failure.

Reply:
You are correct, i changed this and that allowed me to fix the issue related to bypassing the password when clicking back button.